src/Socket.js

```diff
@@ -0,0 +1,109 @@
+import { Socket as PhoenixSocket } from 'phoenix';
+
+export const states = {
+  CONNECTING: 'CONNECTING',
+  CONNECTED: 'CONNECTED',
+  JOINING: 'JOINING',
+  JOINED: 'JOINED',
```

Comment:
I guess an alternative name could be "ready" in that the required number of participants are reached and the game can start. Technically this user has always joined the game himself even at the "JOINING" state, it's just that the others still need to show up.